### ISSUE TITLE
librosa.effects.pitch_shift only accepts mono audio

### ISSUE THREAD

**Issue by l4cr0ss** [librosa/librosa#1085]
#### Description
Invoking `librosa.effects.pitch_shift` with a stereo wave file yields a `ParameterError: Invalid shape for monophonic audio: ndim=2, shape=(163011, 2) `, but passing the `mono=False` flag raises a `TypeError: stft() got an unexpected keyword argument 'mono'`. 

If I write a patch to support stereo pitch shifting will it be accepted? It doesn't have to use the mono kwarg I just see that pattern elsewhere. It could be written to work with N-dim audio. 

#### Steps/Code to Reproduce
<pre><font color="#FCE94F">  6 def</font> <font color="#34E2E2"><b>main</b></font>():                                       
<font color="#FCE94F">  5 </font>    fname = <font color="#AD7FA8">&apos;/home/l4/Music/clips/humming.wav&apos;</font>
<font color="#FCE94F">  4 </font>    y, sr = load_sf(fname)  # function to load the sound file                        
<font color="#FCE94F">  3 </font>    <font color="#34E2E2"><b>print</b></font>(f<font color="#AD7FA8">&quot;shape: {y.shape}, sample rate: {sr}&quot;</font>)            
<font color="#FCE94F">  2 </font>    z = lb.effects.pitch_shift(y, sr, <font color="#AD7FA8">5</font>, mono=<font color="#34E2E2"><b>False</b></font>)         
<font color="#FCE94F">  1 </font>    m = np.column_stack((z.transpose()[<font color="#AD7FA8">0</font>], y.transpose()[<font color="#AD7FA8">1</font>]))
<font color="#FCE94F">127 </font>    play_sa(m,sr) # function to play the wav
<font color="#FCE94F">  1 </font>      
<font color="#FCE94F">  2 </font>main()  </pre>
#### Expected Results
<pre><font color="#FCE94F">  1 </font>soundfile load: /home/l4/Music/clips/humming.wav                           
<font color="#FCE94F">  2 </font>shape: (163011, 2), sample rate: 44100
<font color="#FCE94F">  (audio plays, one channel pitch shifted by 5 semitones)</pre>

#### Actual Results
<pre>Traceback (most recent call last):                                                 
  File &quot;/home/l4/Music/audio.py&quot;, line 129, in &lt;module&gt;                            
    main()                                                                         
  File &quot;/home/l4/Music/audio.py&quot;, line 124, in main                                
    z = lb.effects.pitch_shift(y, sr, 0, mono=False)                               
  File &quot;/home/l4/.local/lib/python3.9/site-packages/librosa/effects.py&quot;, line 314, 
in pitch_shift                                                                     
    y_shift = core.resample(time_stretch(y, rate, **kwargs), float(sr)/rate, sr,   
  File &quot;/home/l4/.local/lib/python3.9/site-packages/librosa/effects.py&quot;, line 235, 
in time_stretch                                                                    
    stft = core.stft(y, **kwargs)                                                  
TypeError: stft() got an unexpected keyword argument &apos;mono&apos;</pre>
#### Versions
<pre>
&gt;&gt;&gt; import platform; print(platform.platform())                                    
Linux-4.15.0-88-lowlatency-x86_64-with-glibc2.27                                                                 
&gt;&gt;&gt; import sys; print(&quot;Python&quot;, sys.version)                                       
Python 3.9.0a4 (default, Feb 26 2020, 03:30:19)                                    
[GCC 7.4.0]                                                                        
&gt;&gt;&gt; import numpy; print(&quot;NumPy&quot;, numpy.__version__)                                
NumPy 1.19.0.dev0+9284251                                                          
&gt;&gt;&gt; import scipy; print(&quot;SciPy&quot;, scipy.__version__)                                
SciPy 1.4.1                                                                        
&gt;&gt;&gt; import librosa; print(&quot;librosa&quot;, librosa.__version__)                          
librosa 0.7.2                                                                      
</pre>

**Comment by lostanlen** [comment]
Yes, i would like that. Librosa is not great at processing multichannel audio currently. But stereo pitch shift seems like a very common application, and it's a bit embarassing that we don't support it as a one-liner.

**Comment by bmcfee** [comment]
Kicking this up a level to #1130
